types/src/test_utils.rs

```diff
@@ -0,0 +1,241 @@
+use tendermint::{
+    block::{
+        header::{Header, Version},
+        Commit, CommitSig,
+    },
+    chain,
+    public_key::PublicKey,
+    Signature, Time,
+};
+
+use crate::block::{CommitExt, GENESIS_HEIGHT};
+use crate::consts::version;
+use crate::hash::{Hash, HashExt};
+use crate::nmt::{NamespacedHash, NamespacedHashExt};
+use crate::{DataAvailabilityHeader, ExtendedHeader, ValidatorSet};
+
+pub type SigningKey = ed25519_consensus::SigningKey;
+
+pub fn generate_ed25519_key() -> SigningKey {
+    SigningKey::new(rand::thread_rng())
+}
+
+impl ExtendedHeader {
```

Comment:
I'm wondering if it wouldn't be more convenient to use if we'd introduce a newtype like `TestExtendedHeader` that would combine the EH and key in itself, so that the key doesn't need to be passed everywhere. It could also `impl Deref<Target = ExtendedHeader>` so it wouldn't be less convenient to use

Reply:
I think If I will add a new type, I prefer `ExtendedHeaderGenerator`. and have the methods there with additional utilities too. Generator will also initialize the SigningKey internally and user doesn't need to know about it.

Reply:
implemented `ExtendedHeaderGenerator`